src/lib.rs

```diff
@@ -675,11 +699,50 @@ fn perform_reductions(png: &mut png::PngData, opts: &Options) -> bool {
         }
     }
 
+    if deadline.passed() {
+        return reduction_occurred;
+    }
+
     png.try_alpha_reduction(&opts.alphas);
 
     reduction_occurred
 }
 
+
+/// Keep track of processing timeout
+struct Deadline {
+    start: Instant,
+    timeout: Option<Duration>,
+    print_message: Mutex<bool>,
```

Comment:
Could this be accomplished with an AtomicBool?

Reply:
Probably, but the implementation would be less obvious. This path is very cold, so it wouldn't make any difference to performance. I'd rather not complicate it.